src/components/KButton/KButton.vue

```diff
@@ -192,23 +204,37 @@ export default {
     cursor: not-allowed;
   }
 
-  // remove pointer events from only <a>
-  &[disabled]:not(:disabled) {
-    pointer-events: none;
-  }
-
   /* Appearances */
 
   &.primary {
-    @include kButtonPrimaryAppearance;
+    background-color: var(--kui-color-background-primary, $kui-color-background-primary);
+    border: var(--kui-border-width-20, $kui-border-width-20) solid var(--kui-color-border-transparent, $kui-color-border-transparent);
+    color: var(--kui-color-text-inverse, $kui-color-text-inverse);
+
+    &:hover:not(:disabled):not([disabled]):not(:focus):not(:active) {
+      background-color: var(--kui-color-background-primary-strong, $kui-color-background-primary-strong);
+    }
+
+    &:focus-visible {
+      background-color: var(--kui-color-background-primary-stronger, $kui-color-background-primary-stronger);
+    }
+
+    &:active {
+      background-color: var(--kui-color-background-primary-strongest, $kui-color-background-primary-strongest);
+    }
+
+    &:disabled, &[disabled] {
+      background-color: var(--kui-color-background-disabled, $kui-color-background-disabled);
+      color: var(--kui-color-text-disabled, $kui-color-text-disabled);
+    }
```

Comment:
We're only moving the primary styles here to avoid having to override all of them for .none. Some of these styles have high specificity, which would make it difficult for the .none styles to take precedence, thus complicating user overrides.

The default size mixin doesn't need to change as we only add style to `.k-button` and will be overriden by a simple `all: unset`.

Reply:
I'm not sure I like moving the primary styles here out of the mixin.

Couldn't we change the mixin implementation to something like this?

```scss
.k-button {
  &:not(.none) {
    @include kButtonPrimaryAppearance;
  }
}
```

Reply:
Actually... 

Now that you modified the buttonAppearance to always include a valid appearance (meaning, if undefined or an invalid appearance is added) I think this is fine as long as the fallback always applies these `primary` styles.

Going to resolve this comment.

Reply:
By moving the styles into `.primary`, they only output once. Do you think there are downsides by doing so?

Reply:
That should be fine. 

The original reason the `.k-button` had the `primary` mixin applied was to ensure that if a user passed an undefined or invalid `appearance` prop, the default `primary` styles would still be applied to the KButton element.

Since you updated the buttonAppearance to be computed and always fallback to `primary` it has the same effect.

cc @portikM